File: api/models/event.go
package models

import (
	"time"

	"gorm.io/datatypes"
)

type EventKind string

const (
	EventKindNodeOffline      EventKind = "node_offline"
	EventKindNodeOnline       EventKind = "node_online"
	EventKindTunnelDown       EventKind = "tunnel_down"
	EventKindTunnelUp         EventKind = "tunnel_up"
	EventKindOSPFNeighborDown EventKind = "ospf_neighbor_down"
	EventKindOSPFNeighborUp   EventKind = "ospf_neighbor_up"
	EventKindIPPoolExhausted  EventKind = "ip_pool_exhausted"
	EventKindNodeDecommission EventKind = "node_decommissioned"
)

type Event struct {
	ID        uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	CreatedAt time.Time `json:"created_at"`

	Kind EventKind `json:"kind" gorm:"not null;index"`

	NodeID *uint `json:"node_id,omitempty" gorm:"index"`
	Node   *Node `json:"node,omitempty" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`

	Message string         `json:"message" gorm:"not null"`
	Data    datatypes.JSON `json:"data,omitempty"`
}

File: web/src/services/types/event.ts
export type EventKind =
  | 'node_offline'
  | 'node_online'
  | 'tunnel_down'
  | 'tunnel_up'
  | 'ospf_neighbor_down'
  | 'ospf_neighbor_up'
  | 'ip_pool_exhausted'
  | 'node_decommissioned';

export interface Event {
  id: number;
  created_at: string;
  kind: EventKind;
  node_id?: number;
  message: string;
  data?: unknown;
}

